### PR TITLE
Add nushell dependency analysis script

### PR DESCRIPTION
**Pull request by tertsdiepraam**
I was playing around with nushell and made a script to analyze our dependencies. I keep trying to get a better overview of all the dependencies that we have and it's a bit of a struggle, because we have so many. I'm planning on writing a little blog post analyzing all our dependencies with this script.

The reason to add this in addition to `cargo tree`, `cargo deny`, etc. is that this gives us a table to analyze in any way we want and extend. See the list of ideas in the comments of the script. We could also, for example, figure out dependants of crates to check they are popular or that we are the only users.

Ultimately, it might be a good idea to go through the entire list of dependencies to figure out whether they are necessary. Or to figure out whether there are multiple crates doing the same thing.

### CODE REVIEW

**Comment by sylvestre** [comment]
TIL about .nu :)

**Comment by sylvestre** [comment]
some typos :)

**Comment by tertsdiepraam** [comment]
Yep, I'll fix them today

File: util/deps.nu [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+#
+# Then you can perform analysis. For example, to group the dependencies by organisation:
+#
+# ```
+# > $dep | group-by organisation   
+# ```
+#
+# Or to find all crates with multiple versions (like cargo deny):
+# ```
+# > $dep | where versions > 1
```
Comment (cakebaker) [inline util/deps.nu:44]:
```suggestion
# > $dep | where num_versions > 1   
```

**Comment by sylvestre** [comment]
can we run it into the CI ?

**Comment by tertsdiepraam** [comment]
Yeah we could if we install nushell. Currently it doesn't give any warnings, it just displays the info, so there's not much use of running it in CI. But we could add things like checking for duplicates and printing the ones that are duplicates for example.

**Comment by tertsdiepraam** [comment]
Bad news by the way: my laptops battery isn't good anymore and has become unsafe to use, so I wont have a computer to develop on for a little while. I'll be back in action at the end of July :)

**Comment by fdncred** [comment]
it's a stretch to say this is tangentially related, but, i saw the word `nushell` so here it goes. 😆  

i've been longing to build `uutils/coreutils` into `nushell`. here's a draft pr i created if anyone is interested in joining forces https://github.com/nushell/nushell/pull/9463

BTW - nice nushell script here! If you decide to run nushell in CI, one of our maintainers created this https://github.com/marketplace/actions/setup-nu

**Comment by tertsdiepraam** [comment]
@fdncred A few weeks back I joined the nushell discord to discuss integrating the two. I've been playing with nushell since then and this is the result. I wanted to contact you all about this once I had more time to work on it. Feel free to open an issue here if you see some way that we can help you with the integration!

Like I said, I can't join the effort before August but I'd love to chat afterwards.

**Comment by fdncred** [comment]
Thanks @tertsdiepraam for the support. Here's an issue I created https://github.com/uutils/coreutils/issues/5088

**Comment by sylvestre** [comment]
@tertsdiepraam do you want to merge this ?

**Comment by tertsdiepraam** [comment]
Yeah I'll clean it up a bit and fix the spelling and then we can merge it. I think this can be quite useful.

**Pull request merged**